src/manager.cpp

```diff
@@ -461,8 +461,27 @@ void Manager::collectSingleFruVpd(
 void Manager::deleteSingleFruVpd(
     const sdbusplus::message::object_path& i_dbusObjPath)
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    try
+    {
+        if (std::string(i_dbusObjPath).empty())
+        {
+            throw std::runtime_error(
+                "Given input is empty. Cannot perform deletion for: " +
```

Comment:
instead of input - can you give as D-bus object path.
and remove appending std::string(i_dbusObjPath)); at line 470 as it is already empty.

Something like
"Given d-bus inventory object path is empty. Aborting FRU VPD deletion."

Reply:
Done